### PR TITLE
Update component to support external events

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

**Fixes:** #11

## Summary:

Scope of this update is to propose a solution for passing a click event into the outer component and passing that event into the scope of the component itself. 

## Type of change:

Please delete options that are not relevant.

- [x] New capability 
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate) 

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._ 

**Thank you for your submission!**<br>
-- Orion Design System Team

### CODE REVIEW

**Comment by znewton** [comment]
This PR confuses me a bit. Perhaps it's the wording of the README I don't understand, or I'm not fully grasping the concept of attributes and event binders in Polymer. My confusion, I think, boils down to 2 main points:
1. In the React example, why does `<button buttonCallback={e => console.log(e)} />` not achieve what the React ref is doing?
2. In a few places, the React ref is referred to as an "event", but then you're setting the buttonCallback on said "event" to a function that takes in an event as the parameter. Is the event parameter a reference to the ref that is an event?  Or is the ref actually the typical ref that is a reference to the DOM Node and its underlying API?

It's possible that neither of those questions made sense, which is a pretty good indicator of how confused I am by this button even work around.

Would using this buttonRef allow me to set other attributes on the shadow dom button? Such as type?

**Comment by znewton** [comment]
It's entirely possible (and likely) that I should play around with this myself and make my own proposal if I come to a different conclusion. I'm just trying to voice my concerns as someone experienced with React who will be using these components heavily in the future.

**Comment by hawkmoth8** [comment]
Dale I already have the HOC done. But I need to talk with you before I moving forward. This might need some updates on the readme.

**Comment by blackfalcon** [comment]
@znewton the CALLBACK.md  has been updated to better communicate the use-case. Please review and please comment directly to the CALLBACK.md file for any further feedback.

File: src/ods-button.js [hunk 1]
```diff
@@ -67,15 +69,19 @@ class OdsButton extends LitElement {
     return disabled || isactive;
   }
 
+  buttonCallback() {
```
Comment (blackfalcon) [inline src/ods-button.js]: This is a default callback within the custom element as to alert the consumer of the component if they have not correctly bound an event to the custom element correctly. 

When the `buttonCallback` prop is overwritten by an external function, this alert will be dismissed.

**Pull request merged**